packages/story-editor/src/components/form/tags/input.js

```diff
@@ -144,16 +244,50 @@ function Input({
           ...renderedTags.slice(renderedTags.length - offset),
         ].map((tag) =>
           tag === INPUT_KEY ? (
-            <TextInput
-              {...props}
-              key={INPUT_KEY}
-              value={value}
-              onKeyDown={handleKeyDown}
-              onChange={handleChange}
-              onFocus={handleFocus}
-              onBlur={handleBlur}
-              size="4"
-            />
+            <Fragment key={id}>
+              <TextInput
+                {...props}
+                key={INPUT_KEY}
+                value={value}
+                id={id}
+                onKeyDown={handleKeyDown}
+                onChange={handleChange}
+                onFocus={handleFocus}
+                onBlur={handleBlur}
+                size="4"
+                ref={inputRef}
+                autoComplete="off"
+              />
+              {filteredAutocompleteSuggestions.length > 0 && value.length >= 3 && (
+                <Popup
+                  anchor={containerRef}
+                  isOpen={filteredAutocompleteSuggestions.length > 0}
+                  placement={dynamicPlacement}
+                  refCallback={dropDownMenuPlacement}
+                  fillWidth
+                >
+                  <MenuWithRef
+                    activeValue={value}
+                    ref={menuRef}
+                    listId={listId}
+                    hasMenuRole
+                    handleReturnToParent={handleReturnFocusToInput}
+                    isRTL={isRTL}
+                    options={[{ group: filteredAutocompleteSuggestions }]}
+                    onMenuItemClick={handleTagSelectedFromSuggestions}
+                    onDismissMenu={noop} // No need to dismiss, it's either open with options or hidden
+                    menuAriaLabel={sprintf(
+                      /* translators: %s: title of freeform terms */
+                      __('Available terms for %s', 'web-stories'),
```

Comment:
Can we use the taxonomy's existing `items_list` label instead? It will be "Tags list" or "Categories list" (or "Colors list", or ...) depending on the taxonomy. Allows us to avoid the generic "term" term.